WPXMLRPCTest/Tests/WPXMLRPCDataCleanerTest.m

```diff
@@ -11,20 +11,61 @@ - (void)testClosingTagRepaired
 {
     WPXMLRPCDataCleaner *cleaner = [[WPXMLRPCDataCleaner alloc] init];
 
+    // Test a whole response
     NSString *completeResponse = @"<methodResponse><params><param><value><string>Hello!</string></value></param></params></methodResponse>";
-    NSString *truncatedResponse = [completeResponse substringWithRange:NSMakeRange(0, [completeResponse length] - 20)];
-    NSString *cleanedString = [cleaner cleanClosingTagIfNeeded:truncatedResponse];
-    STAssertEqualObjects(truncatedResponse, cleanedString, @"Should not try to fix a response that is too damaged.");
+    NSString *cleanedString = [cleaner cleanClosingTagIfNeeded:completeResponse lengthOfCharactersPrecedingPreamble:0];
+    STAssertEqualObjects(completeResponse, cleanedString, @"A good reponse should just be returned intact.");
 
-    completeResponse = @"<methodResponse><params><param><value><string>Hello!</string></value></param></params></methodResponse>";
+    // Test a whole response but some info was cleaned from before the preamble
```

Comment:
Fair point. I considered it but it felt like more boiler plate just to test the different test cases for a single method. Simple to add more methods if it feels easier to maintain.